mail_send_confirmation/static/src/models/composer_view.esm.js

```diff
@@ -0,0 +1,45 @@
+/** @odoo-module **/
+
+import {registerPatch} from "@mail/model/model_core";
+import "@mail/models/composer_view";
+import Dialog from "web.Dialog";
+import core from "web.core";
+const _t = core._t;
+
+registerPatch({
+    name: "ComposerView",
+    recordMethods: {
+        /**
+         * @override
+         */
+        onClickSend() {
+            const composerModel = this;
+            if (this.composer.isLog) {
+                this._super();
+            } else {
+                this.dialog = new Dialog(this, {
+                    title: _t("Confirmation"),
+                    $content: $("<div/>", {
+                        text: _.str.sprintf(
+                            _t(
+                                "This message will be sent to external users as well. Are you sure you want to send this email?"
```

Comment:
```suggestion
                                "This message will be sent to external partners as well. Are you sure you would like to send this message?"
```
Since they are not necessarily users.